## Installation of software in a custom directory

In
```shell
cd /fs/scratch/PAS2658/Alexis/Assembly/Software
```

Installing **BBTools**: a suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data https://jgi.doe.gov/data-and-tools/bbtools/
```shell
wget https://sourceforge.net/projects/bbmap/files/BBMap_39.06.tar.gz
tar -xvzf BBMap_39.06.tar.gz
rm BBMap_39.06.tar.gz
# testing BBTools:
bash ./bbmap/bbmerge.sh
```

Installing **Canu 2.2**: an assembler designed for high-noise single-molecule sequencing (such as the PacBio RS II/Sequel or Oxford Nanopore MinION) https://github.com/marbl/canu
Koren et al. (2017) *Genome Research* doi: https://doi.org/10.1101/gr.215087.116
Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation
Koren S, Rhie A, Walenz BP, Dilthey AT, Bickhart DM, Kingan SB, Hiendleder S, Williams JL, Smith TPL, Phillippy AM. De novo assembly of haplotype-resolved genomes with trio binning. *Nature Biotechnology. (2018)* di: https://doi.org/10.1038/nbt.4277
```shell
pwd
curl -L https://github.com/marbl/canu/releases/download/v2.2/canu-2.2.tar.xz --output canu-2.2.tar.xz
tar -xJf canu-2.2.tar.xz
cd canu-2.2/src
make -j 28
#output Makefile:35: *** git '/usr/bin/git' version '1.8.3.1' too old; at least version 2.12 is required.  Stop. ????


cp -r ../build/bin/ /fs/scratch/PAS2658/Alexis/Assembly/Software/bin/ && cp -r ../build/lib/ /fs/scratch/PAS2658/Alexis/Assembly/Software/lib/ && cp -r ../build/obj/ /fs/scratch/PAS2658/Alexis/Assembly/Software/ && cp -r ../build/share/ /fs/scratch/PAS2658/Alexis/Assembly/Software/share/
cd ../..
# Testing Canu:
/fs/scratch/PAS2658/Alexis/Assembly/Software/bin/canu
```

Installing **Jellyfish 2.3.1**: A fast multi-threaded k-mer counter https://github.com/gmarcais/Jellyfish
```shell
wget https://github.com/gmarcais/Jellyfish/archive/refs/tags/v2.3.1.tar.gz
tar -xvzf jellyfish-2.3.1.tar.gz
cd jellyfish-2.3.1
module load gnu/12.3.0
./configure --prefix=/fs/your/OSC/path/Assembly/Software/
make
make install
cd ..
rm jellyfish-2.3.1.tar.gz
# Testing Jellyfish:
/fs/scratch/PAS2658/Alexis/Assembly/Software/jellyfish-2.3.0/bin/jellyfish
```

Installing **Pilon** 1.24: assembly corrector https://github.com/broadinstitute/pilon/wiki
```shell
wget https://github.com/broadinstitute/pilon/releases/download/v1.24/pilon-1.24.jar
module load java/11.0.8
# Testing Pilon:
java -Xmx125G -jar /fs/scratch/PAS2658/Alexis/Assembly/Software/pilon-1.24.jar
```

Installing **SPAdes 3.15.5**: an assembly toolkit containing various assembly pipelines http://cab.spbu.ru/software/spades/
Bankevich et al. (2012) *Journal of Computation Biology* doi: https://doi.org/10.1089/cmb.2012.0021
SPAdes: A New Genome Assembly Algorithm and Its Applications to Single-Cell Sequencing
```shell
module purge
wget https://github.com/ablab/spades/releases/download/v3.15.5/SPAdes-3.15.5-Linux.tar.gz
tar -xzf SPAdes-3.15.5-Linux.tar.gz
cp /fs/your/OSC/path/Assembly/Software/SPAdes-3.15.5-Linux/bin/* /fs/your/OSC/path/Assembly/Software/bin/
cp -r /fs/your/OSC/path/Assembly/Software/SPAdes-3.15.5-Linux/share/* /fs/your/OSC/path/Assembly/Software/share/
# Testing SPAdes:
/fs/your/OSC/path/Assembly/Software/bin/spades.py --test
rm SPAdes-3.15.5-Linux.tar.gz
```

Installing **Genomescope**: fast genome analysis from unassembled short reads https://github.com/schatzlab/genomescope
Vurture et al. (2017) *Bioinformatics* doi: https://doi.org/10.1093/bioinformatics/btx153
```shell
git clone https://github.com/schatzlab/genomescope.git
# Testing GenomeScope:
module purge
module load gnu/7.3.0
module load mkl/2018.0.3
module load R/3.6.0
Rscript /fs/your/OSC/path/Assembly/Software/genomescope/genomescope.R
```

Installing **nTCard**: estimator of k-mer coverage histogram of genomics data https://github.com/bcgsc/ntCard, *ntCard*: a streaming algorithm for cardinality estimation in genomics data, Mohamadi et al. 2017 doi: https://www.doi.org/10.1093/bioinformatics/btw832, useful for kmergenie
```shell
git clone https://github.com/bcgsc/ntCard.git
cd ntCard/
module purge
module load gnu/9.1.0
./autogen.sh
./configure --prefix=/fs/your/OSC/path/Assembly/Software/ CFLAGS='-g -O3' CXXFLAGS='-g -O3'
make
make install
cd ..
# Testing nTcard:
/fs/your/OSC/path/Assembly/Software/bin/ntcard
```

Installing **KmerGenie 1.7051**: estimator of the best k-mer length for genome de novo assembly http://kmergenie.bx.psu.edu Chikhi et l. 2014 doi: https://doi.org/10.1093/bioinformatics/btt310
```shell
wget http://kmergenie.bx.psu.edu/kmergenie-1.7051.tar.gz
tar -xvzf kmergenie-1.7051.tar.gz
cd kmergenie-1.7051
module load R/3.6.0-gnu7.3
make
make install PREFIX=/fs/your/OSC/path/Assembly/Software/
cd ..
rm kmergenie-1.7051.tar.gz
/fs/your/OSC/path/Assembly/Software/kmergenie-1.7051/kmergenie
```

Installing **assembly-stats**: assembly statistics from FASTA and FASTQ files https://github.com/sanger-pathogens/assembly-stats
```shell
apptainer pull assembly-stats.sif docker://quay.io/biocontainers/assembly-stats:1.0.1--h9f5acd7_5
# Testing Assembly-Stats:
cd ../..
apptainer exec /fs/your/OSC/path/Assembly/Software/assembly-stats.sif assembly-stats
```

Installing **FastQC v0.12.1** https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
```shell
cd /fs/your/OSC/path/Assembly/Software/
apptainer pull fastqc.sif docker://quay.io/biocontainers/fastqc:0.12.1--hdfd78af_0
apptainer exec /fs/your/OSC/path/Assembly/Software/fastqc.sif fastqc --help
cd ..
```
