## Doing K-mer analysis with Jellyfish and GenomeScope

## Ask for computing resources
```
module load perl/5.26.1
sinteractive -c 28 -t 01:00:00 -J Assessment -A PAS2658
```

## Be in the right place
```
cd /fs/scratch/PAS2658/Alexis/Assembly/Illumina/kmer_analysis
```
## Using Jellyfish
```
cp /fs/scratch/PAS2658/Alexis/Assembly/Illumina/Preprocessing/no_duplicates/*.fastq.gz .
# 23-mer
module load gnu/12.3.0 #load a diff Jellyfish so it works
/fs/scratch/PAS2658/Alexis/Assembly/Software/bin/jellyfish count -C <(zcat *.fastq.gz) -m 23 -o Illumina_k23 -s 25G -t 28
/fs/scratch/PAS2658/Alexis/Assembly/Software/bin/jellyfish histo -t 28 -o Illumina_k23.hist Illumina_k23
ls -l #make sure Illumina_k23.hist was created
```
### Obtaining plots and estimations of genome size
```
module purge
module load gnu/7.3.0
module load mkl/2018.0.3
module load R/3.6.0
# For Illumina 23-mer
Rscript /fs/scratch/PAS2658/Alexis/Assembly/Software/genomescope/genomescope.R  /fs/scratch/PAS2658/Alexis/Assembly/Illumina/kmer_analysis/Illumina_k23.hist 23 100 K-mer_s23
```
## A quick way using KmerGenie with the Nanopore data:
```
cd /fs/scratch/PAS2658/Alexis/Assembly/Nanopore/
cp /fs/scratch/PAS2658/Alexis/Assembly/Raw_Data/SRR11481813_1.fastq.gz .
cd kmer_analysis
/fs/scratch/PAS2658/Alexis/Assembly/Software/kmergenie-1.7051/kmergenie Illumina_list.txt -s 2 -t 28 -o Kmergenie_Illumina

```
## Running KmerGenie in Illumina data together
```
cd /fs/scratch/PAS2658/Alexis/Assembly/Illumina/kmer_analysis/ 
# Make a text file listing the names of the fastq.gz files
echo /fs/scratch/PAS2658/Alexis/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_1_clean_nodup.fastq.gz > Illumina_list.txt
echo /fs/scratch/PAS2658/Alexis/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_2_clean_nodup.fastq.gz >> Illumina_list.txt
```
## Request computing power
```
sinteractive -c 28 -t 01:00:00 -J K-kmer_analysis -A PAS2658
```
## Load R
```
module purge
module load gnu/7.3.0
module load mkl/2018.0.3
module load R/3.6.0
```
Run KmerGenie with k-mer size intervals of two, Kmergenie will automatically determine the length of the reads and will start to produce data from 15-mer to the longest k-mer size possible based on the read length
```
mkdir merged_Illumina
cd merged_Illumina
/fs/scratch/PAS2658/Alexis/Assembly/Software/kmergenie-1.7051/kmergenie /fs/scratch/PAS2658/Alexis/Assembly/Illumina/kmer_analysis/Illumina_list.txt -s 2 -t 28 -o Kmergenie_Illumina
# this did not seem to work
```
This command takes us to the path to the executable 'kmergenie'
The Illumina_list.txt is the input file which tells the computer hey do k-mer analysis on these Illumina files

## By gaining this information we can move to try the assembly using both types of data
