The actual assembly of the data will be done in two steps: preliminary assembly of Nanopore data using Canu, and subsequent hybrid assembly using Nanopore assembly (as trusted contigs) and Illumina reads. The actual routines to do the assembly are long (Canu ~15 hours and SPAdes ~1 hours). Thus, we will use the Job Composer of OSC using sequential jobs. Use the scripts below, and copy and paste in the job composer editor.

If you did not finish the k-mer analysis, them make sure of executing the commands below, thus, the Nanopore data will be available for assembly:
```
cd /fs/scratch/PAS2658/Alexis/Assembly/Nanopore/
cp /fs/scratch/PAS2658/Alexis/Assembly/Raw_Data/SRR11481813_1.fastq.gz .
```
## Nanopore assembly batch script

-submitted job 2.28 4:35pm - failed
-changed /bin/canu to canu-2.2, submitted job 2.28 4:40pm
-changed it back. 
- deleted and re-copied file
- -failed again 4:48pm 2.28
-changed canu route and submitted 4.57pm 2.28 - didnt immediately fail. lets hope

- update at 2.29 5:41 am the job finished successfully!

```
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Nanopore_Assembly
#SBATCH --time=24:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2658

export OMP_NUM_THREADS=28
cd /fs/scratch/PAS2658/Alexis/Assembly/Nanopore
# Running Canu
/fs/scratch/PAS2658/Alexis/Assembly/Software/canu-2.2/src/canu-2.2/build/bin/canu useGrid=false -p canu_Assembly -d Nanopore_Assembly genomeSize=19m -nanopore-raw SRR11481813_1.fastq.gz #stopOnLowCoverage=false
```
## Checking the stats from the Nanopore assembly
```
# Using assembly-stats
apptainer exec /fs/scratch/PAS2658/Alexis/Assembly/Software/assembly-stats.sif assembly-stats /fs/scratch/PAS2658/Alexis/Assembly/Nanopore/Nanopore_Assembly/canu_Assembly.contigs.fasta
# Or using BBmap stats
/fs/scratch/PAS2658/Alexis/Assembly/Software/bbmap/stats.sh /fs/scratch/PAS2658/Alexis/Assembly/Nanopore/Nanopore_Assembly/canu_Assembly.contigs.fasta
```
output from apptainer exec ....

stats for /fs/scratch/PAS2658/Alexis/Assembly/Nanopore/Nanopore_Assembly/canu_Assembly.contigs.fasta
sum = 19104297, n = 6, ave = 3184049.50, largest = 4883604
N50 = 3672067, n = 3
N60 = 3672067, n = 3
N70 = 3519523, n = 4
N80 = 3519523, n = 4
N90 = 2767752, n = 5
N100 = 138539, n = 6
N_count = 0
Gaps = 0



Once the assembly using Nanopore has been finished, the resulted contigs can be used as trusted contigs for hybrid assembly with SPAdes.

We prepare the Illumina reads (they have quality already, just we need them ready for processing):

## Verify they are paired
````
cd /fs/scratch/PAS2658/Alexis/Assembly/Joint_Assembly/
/fs/scratch/PAS2658/Alexis/Assembly/Software/bbmap/reformat.sh in1=/fs/scratch/PAS2658/Alexis/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_1_clean_nodup.fastq.gz in2=/fs/scratch/PAS2658/Alexis/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_2_clean_nodup.fastq.gz verifypairing
````
## We correct them to make them paired
```
/fs/scratch/PAS2658/Alexis/Assembly/Software/bbmap/reformat.sh in1=/fs/scratch/PAS2658/Alexis/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_1_clean_nodup.fastq.gz in2=/fs/scratch/PAS2658/Alexis/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_2_clean_nodup.fastq.gz out1=paired_Illumina_reads_1.fastq.gz out2=paired_Illumina_reads_2.fastq.gz
```
## The reads need to be renamed
```
/fs/scratch/PAS2658/Alexis/Assembly/Software/bbmap/bbrename.sh in1=paired_Illumina_reads_1.fastq.gz in2=paired_Illumina_reads_2.fastq.gz out1=Illumina_reads_1.fastq.gz out2=Illumina_reads_2.fastq.gz
```
output

Input is being processed as paired
Input:                          50731958 reads                  5038928350 bases
Output:                         50731958 reads (100.00%)        5038928350 bases (100.00%)
Time:                           134.558 seconds.
Reads Processed:      50731k    377.03k reads/sec
Bases Processed:       5038m    37.45m bases/sec


## Using SPAdes takes also long, then it is recommended to use the following script for the OSC Batch system.

submitted 2.29 10:42 am
failed instantly
investigated file names, found inconsistency
edited Illumina reads file name to paired Illumina reads file name and submitted 10:44am 2.29
did not instantly fail. crossing fingers

```
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name SPAdes_assembly
#SBATCH --time=24:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2658

export OMP_NUM_THREADS=28
cd /fs/scratch/PAS2658/Alexis/Assembly/Joint_Assembly
/fs/scratch/PAS2658/Alexis/Assembly/Software/bin/spades.py --isolate -1 /fs/scratch/PAS2658/Alexis/Assembly/Joint_Assembly/paired_Illumina_reads_1.fastq.gz -2 /fs/scratch/PAS2658/Alexis/Assembly/Joint_Assembly/paired_Illumina_reads_2.fastq.gz --trusted-contigs /fs/scratch/PAS2658/Alexis/Assembly/Nanopore/Nanopore_Assembly/canu_Assembly.contigs.fasta -t 28 -m 128 --phred-offset 33 -o SPAssembly
```
If you want to run the analysis with additional k-mer values, you can use the option -k with odd numbers separated by commas, e.g. ```-k 35,37,41```

Checking the stats of the Nanopore + Illumina assembly
```
# Using assembly-stats
apptainer exec /fs/scratch/PAS2658/Alexis/Assembly/Software/assembly-stats.sif /fs/scratch/PAS2658/Alexis/Assembly/Nanopore/Nanopore_Assembly/Joint_Assembly/SPAssembly/scaffolds.fasta #this did not work
# Or using BBmap stats
/fs/scratch/PAS2658/Alexis/Assembly/Software/bbmap/stats.sh /fs/scratch/PAS2658/Alexis/Assembly/Joint_Assembly/SPAssembly/scaffolds.fasta
```
BBmap stats output

A       C       G       T       N       IUPAC   Other   GC      GC_stdev
0.2404  0.2589  0.2596  0.2411  0.0000  0.0000  0.0000  0.5185  0.1070
Main genome scaffold total:             408
Main genome contig total:               412
Main genome scaffold sequence total:    19.086 MB
Main genome contig sequence total:      19.086 MB       0.002% gap
Main genome scaffold N/L50:             3/3.15 MB
Main genome contig N/L50:               3/3.15 MB
Main genome scaffold N/L90:             7/491.45 KB
Main genome contig N/L90:               7/477.184 KB
Max scaffold length:                    4.548 MB
Max contig length:                      4.548 MB
Number of scaffolds > 50 KB:            11
% main genome in scaffolds > 50 KB:     99.08%
Minimum         Number          Number          Total           Total           Scaffold
Scaffold        of              of              Scaffold        Contig          Contig  
Length          Scaffolds       Contigs         Length          Length          Coverage
--------        --------------  --------------  --------------  --------------  --------
    All                    408             412      19,086,256      19,085,856   100.00%
     50                    408             412      19,086,256      19,085,856   100.00%
    100                    370             374      19,083,527      19,083,127   100.00%
    250                    234             238      19,054,917      19,054,517   100.00%
    500                     67              71      19,000,137      18,999,737   100.00%
   1 KB                     26              30      18,972,183      18,971,783   100.00%
 2.5 KB                     15              19      18,957,597      18,957,197   100.00%
   5 KB                     14              18      18,953,144      18,952,744   100.00%
  10 KB                     12              16      18,942,140      18,941,740   100.00%
  25 KB                     12              16      18,942,140      18,941,740   100.00%
  50 KB                     11              15      18,911,145      18,910,745   100.00%
 100 KB                     10              14      18,829,281      18,828,881   100.00%
 250 KB                     10              14      18,829,281      18,828,881   100.00%
 500 KB                      6               7      17,093,697      17,093,597   100.00%
   1 MB                      5               5      16,171,594      16,171,594   100.00%
 2.5 MB                      4               4      13,892,118      13,892,118   100.00%
