The actual assembly of the data will be done in two steps: preliminary assembly of Nanopore data using Canu, and subsequent hybrid assembly using Nanopore assembly (as trusted contigs) and Illumina reads. The actual routines to do the assembly are long (Canu ~15 hours and SPAdes ~1 hours). Thus, we will use the Job Composer of OSC using sequential jobs. Use the scripts below, and copy and paste in the job composer editor.

If you did not finish the k-mer analysis, them make sure of executing the commands below, thus, the Nanopore data will be available for assembly:
```
cd /fs/your/OSC/path/Assembly/Nanopore/
cp /fs/your/OSC/path/Assembly/Raw_Data/SRR11481813_1.fastq.gz .
```
## Nanopore assembly batch script
```
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Nanopore_Assembly
#SBATCH --time=24:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

export OMP_NUM_THREADS=28
cd /fs/your/OSC/path/Assembly/Nanopore
# Running Canu
/fs/your/OSC/path/Assembly/Software/bin/canu useGrid=false -p canu_Assembly -d Nanopore_Assembly genomeSize=19m -nanopore-raw SRR11481813_1.fastq.gz #stopOnLowCoverage=false
```
## Checking the stats from the Nanopore assembly
```
# Using assembly-stats
apptainer exec /fs/your/OSC/path/Assembly/Software/assembly-stats.sif assembly-stats /fs/your/OSC/path/Assembly/Nanopore/Nanopore_Assembly/canu_Assembly.contigs.fasta
# Or using BBmap stats
/fs/your/OSC/path/Assembly/Software/bbmap/stats.sh /fs/your/OSC/path/Assembly/Nanopore/Nanopore_Assembly/canu_Assembly.contigs.fasta
```
Once the assembly using Nanopore has been finished, the resulted contigs can be used as trusted contigs for hybrid assembly with SPAdes.

We prepare the Illumina reads (they have quality already, just we need them ready for processing):

## Verify they are paired
````
cd /fs/your/OSC/path/Assembly/Joint_Assembly/
/fs/your/OSC/path/Assembly/Software/bbmap/reformat.sh in1=/fs/your/OSC/path/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_1_clean_nodup.fastq.gz in2=/fs/your/OSC/path/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_2_clean_nodup.fastq.gz verifypairing
````
## We correct them to make them paired
```
/fs/your/OSC/path/Assembly/Software/bbmap/reformat.sh in1=/fs/your/OSC/path/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_1_clean_nodup.fastq.gz in2=/fs/your/OSC/path/Assembly/Illumina/Preprocessing/no_duplicates/SRR2188474_2_clean_nodup.fastq.gz out1=paired_Illumina_reads_1.fastq.gz out2=paired_Illumina_reads_2.fastq.gz
```
## The reads need to be renamed
```
/fs/your/OSC/path/Assembly/Software/bbmap/bbrename.sh in1=paired_Illumina_reads_1.fastq.gz in2=paired_Illumina_reads_2.fastq.gz out1=Illumina_reads_1.fastq.gz out2=Illumina_reads_2.fastq.gz
```

## Using SPAdes takes also long, then it is recommended to use the following script for the OSC Batch system.
```
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name SPAdes_assembly
#SBATCH --time=24:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

export OMP_NUM_THREADS=28
cd /fs/your/OSC/path/Assembly/Joint_Assembly
/fs/your/OSC/path/Assembly/Software/bin/spades.py --isolate -1 /fs/your/OSC/path/Assembly/Joint_Assembly/Illumina_reads_1.fastq.gz -2 /fs/your/OSC/path/Assembly/Joint_Assembly/Illumina_reads_2.fastq.gz --trusted-contigs /fs/your/OSC/path/Assembly/Nanopore/Nanopore_Assembly/canu_Assembly.contigs.fasta -t 28 -m 128 --phred-offset 33 -o SPAssembly
```
If you want to run the analysis with additional k-mer values, you can use the option -k with odd numbers separated by commas, e.g. ```-k 35,37,41```

Checking the stats of the Nanopore + Illumina assembly
```
# Using assembly-stats
/fs/your/OSC/path/Assembly/Software/assembly-stats/assembly-stats /fs/your/OSC/path/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta
# Or using BBmap stats
/fs/your/OSC/path/Assembly/Software/bbmap/stats.sh /fs/your/OSC/path/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta
```
